src/unix/linux_like/mod.rs

```diff
@@ -12,6 +12,53 @@ extern_ty! {
     pub enum timezone {}
 }
 
+cfg_if! {
+    if #[cfg(not(any(
+        target_env = "musl",
+        target_os = "emscripten",
+        target_env = "ohos"
+    )))] {
+        s! {
+            pub struct sched_param {
+                pub sched_priority: c_int,
+            }
+        }
+    } else if #[cfg(musl_v1_2_3)] {
+        s! {
+            struct __c_anon_sched_param__reserved2 {
+                __reserved1: crate::time_t,
+                __reserved2: c_long,
+            }
+
+            pub struct sched_param {
+                pub sched_priority: c_int,
+
+                __reserved1: c_int,
+                #[cfg(musl32_time64)]
+                __reserved2: [c_long; 4],
+                #[cfg(not(musl32_time64))]
+                __reserved2: [__c_anon_sched_param__reserved2; 2],
+                __reserved3: c_int,
+            }
+        }
+    } else {
+        s! {
+            pub struct sched_param {
+                pub sched_priority: c_int,
+
+                #[deprecated(since = "0.2.173", note = "This field has been removed upstream")]
+                pub sched_ss_low_priority: c_int,
+                #[deprecated(since = "0.2.173", note = "This field has been removed upstream")]
+                pub sched_ss_repl_period: crate::timespec,
+                #[deprecated(since = "0.2.173", note = "This field has been removed upstream")]
+                pub sched_ss_init_budget: crate::timespec,
+                #[deprecated(since = "0.2.173", note = "This field has been removed upstream")]
+                pub sched_ss_max_repl: c_int,
+            }
+        }
+    }
+}
```

Comment:
Commit "musl: time64: update struct sched_param": to keep things a bit cleaner, move these musl definitions to the `musl` module. Then keep the existing `struct sched_param` in this module but add the deprecated notes and move it to a `cfg_if` block with the `any(musl, emscripten, ohos)` bit.

Also could you put this `cfg_if!` after the main `s!` macro with the others? Around line 252 looking at this file currently.